Assets/Mirror/Examples/AdditiveScenes/Scripts/PlayerController.cs

```diff
@@ -5,28 +5,26 @@ namespace Mirror.Examples.Additive
     [RequireComponent(typeof(CharacterController))]
     public class PlayerController : NetworkBehaviour
     {
-        public override void OnStartServer()
-        {
-            base.OnStartServer();
-            playerColor = Random.ColorHSV(0f, 1f, 1f, 1f, 0.5f, 1f);
-        }
-
-        [SyncVar(hook = nameof(SetColor))]
-        public Color32 playerColor = Color.black;
+        public CharacterController characterController;
 
-        // Unity clones the material when GetComponent<Renderer>().material is called
-        // Cache it here and destroy it in OnDestroy to prevent a memory leak
-        Material cachedMaterial;
-
-        void SetColor(Color32 color)
+        void OnValidate()
         {
-            if (cachedMaterial == null) cachedMaterial = GetComponent<Renderer>().material;
-            cachedMaterial.color = color;
+            if (characterController == null)
+            {
+                // Auto-assign CharacterController.
+                characterController = GetComponent<CharacterController>();
+                if (characterController == null)
```

Comment:
I don't think it is necessary to add the component in validate because the [RequireComponent] on the top does this

Reply:
removed